src/presets/firebase/preset.ts

```diff
@@ -33,6 +33,12 @@ const firebase = defineNitroPreset(
         nitro.options.appConfig.nitro = nitro.options.appConfig.nitro || {};
         nitro.options.appConfig.nitro.firebase = nitro.options.firebase;
 
+        if(nitro.options.firebase?.serverFunctionName.includes('-')){
+          nitro.logger.warn(
```

Comment:
I believe it is more safe that we throw an error here as it is a wrong build with changing.